### PR TITLE
Cleaner construction of defaults in MongoNav

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Using lodash `defaultsDeep` to construct default URLs in MongoNav

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: packages/mongo-nav/package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "@leafygreen-ui/badge": "^3.0.0",
     "@leafygreen-ui/button": "^4.0.0",
     "@leafygreen-ui/hooks": "^2.0.0",
-    "@leafygreen-ui/icon-button": "^4.0.0"
+    "@leafygreen-ui/icon-button": "^4.0.0",
+    "lodash": "^4.17.15"
```
Comment (DesignerDave) [inline packages/mongo-nav/package.json]: Can we make sure that lodash is a dependency wherever we use it, and then add it to our webpack externals? I think most applications already install it, so there's not much of a need to bundle it.

File: packages/mongo-nav/src/MongoNav.tsx [hunk 2]
```diff
@@ -121,110 +122,70 @@ export default function MongoNav({
...
 
-  const defaultOrgURL = (orgId: string) =>
-    `${cloudHost}/v2#/org/${orgId}/projects`;
-  const constructOrganizationURL =
-    constructOrganizationURLProp ?? defaultOrgURL;
-
-  const defaultProjectURL = (projectId: string) =>
-    `${cloudHost}/v2#/${projectId}`;
-  const constructProjectURL = constructProjectURLProp ?? defaultProjectURL;
+  const sanitizedHosts = defaultsDeep(hosts, defaultHosts);
```
Comment (DesignerDave) [inline packages/mongo-nav/src/MongoNav.tsx]: Still could probably use a better name than `sanitizedX`...

File: packages/mongo-nav/src/MongoNav.tsx [hunk 3]
```diff
@@ -121,110 +122,70 @@ export default function MongoNav({
...
-  const supportHost = hosts?.support ?? `https://support.mongodb.com`;
-
-  const sanitizedHosts: Required<HostsInterface> = {
-    account: accountHost,
-    cloud: cloudHost,
-    university: universityHost,
-    support: supportHost,
-    charts: hosts?.charts ?? `https://charts.mongodb.com`,
-    stitch: hosts?.stitch ?? `https://stitch.mongodb.com`,
+  const defaultHosts: Required<HostsInterface> = {
```
Comment (hswolff) [inline packages/mongo-nav/src/MongoNav.tsx]: Can move out of the component so we don't redefine it every time.

File: packages/mongo-nav/src/MongoNav.tsx [hunk 4]
```diff
@@ -121,110 +122,70 @@ export default function MongoNav({
...
 
-  const defaultOrgURL = (orgId: string) =>
-    `${cloudHost}/v2#/org/${orgId}/projects`;
-  const constructOrganizationURL =
-    constructOrganizationURLProp ?? defaultOrgURL;
-
-  const defaultProjectURL = (projectId: string) =>
-    `${cloudHost}/v2#/${projectId}`;
-  const constructProjectURL = constructProjectURLProp ?? defaultProjectURL;
+  const constructedHosts = defaultsDeep(hosts, defaultHosts);
```
Comment (hswolff) [inline packages/mongo-nav/src/MongoNav.tsx]: optional: Perhaps rename the internal value of the prop so it's `({ hosts = hostsProps })` and then just name this variable `hosts`.

Ditto for URLs.

File: packages/mongo-nav/src/MongoNav.tsx (hunk 3 above)
Comment (brookescarlett) [reply packages/mongo-nav/src/MongoNav.tsx]: 👍

File: packages/mongo-nav/src/MongoNav.tsx (hunk 4 above)
Comment (brookescarlett) [reply packages/mongo-nav/src/MongoNav.tsx]: 👍

**Pull request merged**